### DISCUSSION TITLE
Redirect to root page with data not possible?

### DISCUSSION THREAD

**Discussion by hazzo** [remix-run/remix#2929] [Q&A] [answered]
Hi all, 

I was looking to make a redirect from a loader in a route to the root page with some data. But data seems to get lost in the redirect,  as it arrives with null value in the body.
It's possible to do this?

```ts
// some_page.ts
export const loader: LoaderFunction = async ({ request }) => {
    const userLogged = await isUserLogged(request);

  if (!userLogged) {
    const locale = detectLocale({ url: new URL(request.url) });
    return new Response(JSON.stringify({ protectedRouteAccessAttempt: true }), {
      status: 302,
      headers: {
        Location: `/`
      }
    });
  }
  return null;
};

// root.ts
export const loader: LoaderFunction = async ({ request }) => {
    console.log(request.body) // null
    return null;
};

```

**Comment by sergiodxa** [comment, marked as answer]
Two things are happening here.

1. Redirects don't pass the body to the target location.
2. Loaders only handle GET requests and GET request don't have a body, a body is only available for an non GET request like POST/PUT/PATCH/DELETE

There're two way you could pass data from a redirect to another loader.

1. Store them in a cookie/session and read from there in the loader you redirected to.
2. Add them to the Location URL as search params and read in the loader from there.

**Reply by hazzo** [reply]
Thanks @sergiodxa. Yes I understand. I finally did it as query param to the redirect url.
